[ADD] module mrp_produce_uos

tafaRU opened the pull request:
This module adds the field uos quantity to Product Produce wizard allowing the user to use the uos quantity (instead of the standard uom) to specify the quantity to be produced.

tafaRU committed:
Add module mrp_produce_uos

eLBati commented:
:+1: thanks

archetipo commented:
:+1:

tafaRU commented:
Could someone of @OCA/manufacturing-maintainers merge it?

Thank you in advance.

tafaRU commented:
Could someone of @OCA/manufacturing-maintainers please review and possibly merge this PR?

Thank you in advance!

tafaRU commented:
@pedrobaeza @max3903 coul you please please review and possibly merge this PR?

Thank you in advance!

pedrobaeza commented on mrp_produce_uos/models/mrp_production.py:
```diff
@@ -0,0 +1,32 @@
+            product_uos_qty = p_uos_qty * (product_uom_qty / p_qty)
+            # I used sql to avoid
+            # https://github.com/odoo/odoo/blob/8.0/addons/stock/stock.py#L1980
+            self.env.cr.execute(
```
You can still use directly `models.write()` for not avoiding security rules or the recalculation of computed fields.

tafaRU replied:
The link in the meantime has been modified and got no clues on the new one.
I've tried but I remember it causes a loop. What I did seems to be the only way.

pedrobaeza replied:
Try what I propose that I think it will work for sure.

tafaRU replied:
Tried with:
`move.sudo().write({'product_uos_qty': product_uos_qty})
`

but I get the following error: https://github.com/odoo/odoo/blob/8.0/addons/stock/stock.py#L2008

Any hints how to fix it?

tafaRU committed:
Fix to avoid a possible ZeroDivisionError

tafaRU committed:
Avoid models folder since there's no need to update product_uos_qty on move_created_ids2.
This is already done by split method of stock.move (see file stock/stock.py as reference) updating product_uos_qty on move_created_ids

tafaRU commented:
@pedrobaeza, I've addressed your last remark with https://github.com/OCA/manufacture/commit/e8829848191717c4f265c2ff512bb2840d387477 :wink: 

Thanks a lot for your review! I think now it should be mergeable, what do you think about it?

pedrobaeza commented:
:+1:
